CONTRIBUTING.md

```diff
@@ -1,75 +1,389 @@
-line like `<<< @/snippets/test-intro-zoe.js#install` to your document.
-The section `test-intro-zoe.js` should be replaced with your
-particular filename. `#install` is an example of a specific code
-region in the file to import. For example, it might look like:
+Code snippets are not short inline code bits like `const x = 2 + 2;`. Or 
+code blocks denoted in Markdown by starting with a line consisting of three backquotes with 
+`js` appended and ending with a line consisting of three backquotes. 
 
+Rather, code snippets are actual development or test code from the Agoric-SDK repo,
+or code held to a similar standard of correctness. They should pass `lint` and run with no
+errors. This ensures our documents use real code that works
+with the current agoric-sdk version (whatever is on master) and is
+not outdated. **tyg todo: I'm not quite seeing this guaranteeing things,
+since we're not extracting the code directly from agoric-sdk. It'd seem
+if we don't check and manually update the snippets file after each upgrade, we risk
+falling out of sync with the agoric-sdk repo.**
+
+To import code snippets into the documentation, create a file
+under `Agoric/documentation/snippets/`. **tyg todo: It appears there should be only one 
+snippets folder for the whole doc repo?** Put your code in the new file.
+Remember, it must be able to run without errors. 
+
+You can make an entire code file, or any part of it, into a snippet (each
+of which can be used multiple times in the docs). Just surround the part
+you want to be a snippet with `#region` and `#endregion` comments:
 ```js
-  // #region install
-  const atomicSwapInstallation = await E(zoe).install(atomicSwapBundle);
-  // #endregion install
+// #region regionName
+...
+// #endregion regionName
 ```
+`regionName` in the above is any name you want to give this snippet.
+
+If you want the whole file to be a snippet, just put the `#region` / `#endregion`
+pair at the start and end of the file. You can define any number of snippet
+regions in a file, including defining one region inside of another. Just be sure to give
+all the ones in a file different names. **tyg todo: Is this all right?**
+
+To include a defined snippet in a Markdown file, put a
+line like `<<< @/snippets/test-intro-zoe.js#install` in it..
+Replace the `test-intro-zoe.js` with the filename in the snippets file.
+Replace the `install` with the name of the region you want included from
+the file. 
 
-Write tests using AVA, and then run them with `yarn test`. The code
-files can also be linted with `yarn lint-fix`. Testing code snippets
-allows us to ensure that our documents are using real code that works
-with the current version of agoric-sdk (whatever is on master) and is
-not outdated.
+To test your snippets files:
+1. Write tests using AVA.
+2. Run the tests with `yarn test`. **tyg todo: (Do all the tests have to be in the snippets directory? And do you run `yarn test` from there?)**
```

Comment:
The tests should be in the snippets directory. Actually the snippet files and the test files ended up being the same file because that was the cleanest way to go. The snippet files should all be named `test-original-filename.js` where `original-filename` is the markdown filename, like `amounts` of `ertp/guide/amounts.md`

Then, running `yarn test` anywhere will run the tests.

Reply:
Not sure if I'm following you, since I've not used or looked at AVA. Does this mean a single file test-original-filename.js both has AVA test(s) and code snippet(s) [the latter denoted by #region and #endregion markers] in it? With the tests specifically testing the code snippet and that snippet only?

Reply:
Yup, you can take a look at https://github.com/Agoric/documentation/blob/6763ee64d5fc66c80a7e3fb4c7a680c41db3a73e/snippets/ertp/guide/test-amounts.js and the other files on that branch to see how it's organized.